tt_metal/hw/ckernels/blackhole/metal/llk_api/llk_sfpu/llk_math_eltwise_unary_sfpu_activations.h

```diff
@@ -24,4 +24,23 @@ inline void llk_math_eltwise_unary_sfpu_hardsigmoid(uint dst_index, int vector_m
         dst_index,
         vector_mode);
 }
+
+template <bool APPROXIMATE>
+inline void llk_math_eltwise_unary_sfpu_celu_init() {
+    llk_math_eltwise_unary_sfpu_init<SfpuType::celu, APPROXIMATE>(
+        ckernel::sfpu::_init_exponential_<APPROXIMATE, /*FAST_APPROX=*/APPROXIMATE, /*SCALE=*/p_sfpu::kCONST_1_FP16B>);
+}
+
+template <bool APPROXIMATE, ckernel::ActivationType ACTIVATION, int ITERATIONS = 8>
+inline void llk_math_eltwise_unary_sfpu_celu(
+    uint dst_index, uint32_t alpha, uint32_t alpha_recip, int vector_mode = (int)VectorMode::RC) {
+    _llk_math_eltwise_unary_sfpu_params_<APPROXIMATE>(
+        [](uint32_t alpha, uint32_t alpha_recip) {
+            ckernel::sfpu::_calculate_activation_<APPROXIMATE, ACTIVATION, ITERATIONS>(alpha, alpha_recip);
+        },
```

Comment:
Why do we need a lambda instead of the function pointer directly? i.e.
```suggestion
        ckernel::sfpu::_calculate_activation_<APPROXIMATE, ACTIVATION, ITERATIONS>,
```